pkg/agent/core/state.go

```diff
@@ -995,6 +1118,11 @@ func (h PluginHandle) RequestSuccessful(now time.Time, resp api.PluginResponse)
 	// the process of moving the source of truth for ComputeUnit from the scheduler plugin to the
 	// autoscaler-agent.
 	h.s.Plugin.Permit = &resp.Permit
+	revsource.Propagate(now,
+		targetRevision,
+		&h.s.Plugin.CurrentRevision,
+		h.s.Config.PromMetricsCallbacks.PluginLatency,
+	)
```

Comment:
Reading through the related code — will the plugin latency get recorded as:

1. end-to-end time from decision to plugin request completed; or
2. latency between end of previous request to end of plugin request

?

OTOH I thought from reading that it was (1), and I would've expected (2). But I'm not sure if I missed something

Reply:
We should ideally have 2 metrics: one with retries, and one without. It is sufficient to have only one in this PR.

I'll think how to best implement it.

Reply:
Plugin/NeonVM/Monitor metrics we currently have do not count retries, i. e. measure each individual request separately. 

In order to count the retries also, we must not do `TargetRevision: s.TargetRevision.WithTime(now)` every time a request is planned. Rather, we update the time once per concrete revision value, and not update it if more request happen with the same revision.